src/main/java/com/synopsys/integration/detect/Application.java

```diff
@@ -81,8 +82,17 @@ public static void setShouldExit(boolean shouldExit) {
     public static void main(String[] args) {
         SpringApplicationBuilder builder = new SpringApplicationBuilder(Application.class);
         builder.logStartupInfo(false);
-        ApplicationUpdater updater = new ApplicationUpdater(new ApplicationUpdaterUtility(), args);
-        if (!updater.selfUpdate()) {
+        boolean selfUpdated = false;
+        ApplicationUpdater updater = null;
+        try {
+            updater = new ApplicationUpdater(new ApplicationUpdaterUtility(), args);
```

Comment:
Couldn't complete my review yesterday though all functional changes look good to me, thanks for the PR comments!
A note to add here is that even try-finally gets caught by SonarCloud as a "Critical" code smell so we can resort to always use try-with-resources to avoid reducing our score. Can we correct this in #807 opened recently?
Sonar Cloud Issue: https://sonarcloud.io/project/issues?resolved=false&severities=CRITICAL&sinceLeakPeriod=true&types=CODE_SMELL&id=com.synopsys.integration%3Asynopsys-detect&open=AYgE3mpW8AAkg9tAaidT